main.go

```diff
@@ -0,0 +1,135 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+type Competitor struct {
+	Nr int
+	Pi int
+}
+
+type Pair struct {
+	fNr   int
+	sNr   int
+	delta int
+}
+
+/*
+|--------------------------------------------------------------------------
+| Task description
+|--------------------------------------------------------------------------
+|
+| The Goal
+| Casablanca’s hippodrome is organizing a new type of horse racing: duals. During a dual,
+| only two horses will participate in the race.
+| In order for the race to be interesting,
+| it is necessary to try to select two horses with similar strength.
+|
+| Write a program which, using a given number of strengths,
+| identifies the two closest strengths and shows their difference with an integer (≥ 0).
+|
+| Input
+| Line 1: Number N of horses
+|
+| The N following lines: the strength Pi of each horse. Pi is an integer.
+|
+| Output
+| The difference D between the two closest strengths. D is an integer greater than or equal to 0.
+|
+*/
+func main() {
+	input, err := getUserInput()
+
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+
+	// make separate slice for the keep competitor Nrs
+	// will use competitor Nrs for pretty look output
+	competitors := make([]Competitor, len(input))
```

Comment:
I would recommend making it better by using `competitors := make([]Competitor, 0, len(input))` which allocates memory only for the capacity. This way, no Competitor object will be created and you will create it in the loop. This means fewer object creation operations, and in the loop, you can call 'append' instead of using the index.